scenario_player/runner.py

```diff
@@ -153,279 +256,260 @@ def local_seed(self) -> str:
-            discovered = self.wait_for_token_network_discovery(node.base_url)
-            log.info("Token Network Discovery", node=node._index, network=discovered)
-        assert discovered
-        return discovered
+        log.info("Setup done, running scenarion", token_network_address=token_network_address)
 
-    def _setup(self, mint_gas, node_count, node_addresses, fund_tx, node_starter, greenlets):
-        ud_token_tx, udc_ctr, should_deposit_ud_token = self._initialize_udc(
-            gas_limit=mint_gas, node_count=node_count
-        )
+        # Start root task
+        self.root_task()
 
-        mint_tx = self._initialize_scenario_token(
-            node_addresses=node_addresses,
-            udc_ctr=udc_ctr,
-            should_deposit_ud_token=should_deposit_ud_token,
-            gas_limit=mint_gas,
-        )
+    def setup_raiden_nodes_ether_balances(self, node_addresses) -> Set[TransactionHash]:
+        """ Makes sure every Raiden node has at least `NODE_ACCOUNT_BALANCE_MIN`. """
 
-        wait_for_txs(self.client, fund_tx | ud_token_tx | mint_tx)
+        ether_transfer_transactions: Set[TransactionHash] = set()
 
-        if node_starter is not None:
-            log.debug("Waiting for nodes to finish starting")
-            node_starter.get(block=True)
+        for address in node_addresses:
+            balance = self.client.balance(to_canonical_address(address))
 
-        first_node = self.get_node_baseurl(0)
+            if balance < NODE_ACCOUNT_BALANCE_MIN:
+                eth_transfer = EthTransfer(
+                    to_address=to_canonical_address(address),
+                    value=NODE_ACCOUNT_BALANCE_FUND - balance,
+                    gas_price=self.client.web3.eth.gasPrice,
+                )
+                tx_hash = self.client.transact(eth_transfer)
+                ether_transfer_transactions.add(tx_hash)
 
-        registered_tokens = set(
-            self.session.get(
-                API_URL_TOKENS.format(protocol=self.protocol, target_host=first_node)
-            ).json()
-        )
-        if self.token.checksum_address not in registered_tokens:
-            for _ in range(5):
-                code, msg = self.register_token(self.token.checksum_address, first_node)
-                if 199 < code < 300:
-                    break
-                # FIXME: Remove long sleep and instead intelligently handle Raiden 5 block wait
-                gevent.sleep(15)
-            else:
-                log.error("Couldn't register token with network", code=code, message=msg)
-                raise TokenRegistrationError(msg)
-
-        self.token_network_address = self.ensure_token_network_discovery()
-        log.info(
-            "Token Network Discovery Completed", token_network_address=self.token_network_address
-        )
-        log.info("Setup done")
-        # Start root task
-        root_task_greenlet = gevent.spawn(self.root_task)
```

Comment:
This extra greenlet is unecessary, calling `self.root_task` has the same effect.